custom_components/octopus_intelligent/octopus_intelligent_system.py

```diff
@@ -48,7 +48,7 @@ async def _async_update_data(self):
         try:
             # Note: asyncio.TimeoutError and aiohttp.ClientError are already
             # handled by the data update coordinator.
-            async with async_timeout.timeout(30):
+            async with async_timeout.timeout(60):
```

Comment:
How about something along the following lines (may need amending):
```suggestion
            timeout = max(30, min(300, update_interval / 2))
            async with async_timeout.timeout(timeout):
```
Where `update_interval` is the value from the `__init__` method (maybe `self.update_interval`, not sure), which used to be 5 minutes and this PR currently proposes amending. The intention being: The timeout should be half the update interval, capped at a minimum of 30s and a maximum of 300s (5 minutes).

Rationale: Given that this integration runs 24/7 and automations depend on it, as a user I would rather face request timeouts as crazy long as, say, 5 minutes rather than facing “unavailable” errors.

If the update interval was configurable by the user as suggested in another comment, the 30s minimum timeout would imply a minimum polling interval of 60s.